src/arithmetic/curves.rs

```diff
@@ -28,12 +25,11 @@ pub trait CurveExt:
     + ConditionallySelectable
     + ConstantTimeEq
     + From<<Self as PrimeCurve>::Affine>
-    + Group<Scalar = <Self as group::Group>::Scalar>
 {
```

Comment:
On line 67, consider disabling the type_complexity lint.
```diff
@@ -68,6 +68,7 @@ pub trait CurveExt:
     ///     (g * x + &(h * r)).to_affine()
     /// }
     /// ```
+    #[allow(clippy::type_complexity)]
     fn hash_to_curve<'a>(domain_prefix: &'a str) -> Box<dyn Fn(&[u8]) -> Self + 'a>;
 
     /// Returns whether or not this element is on the curve; should
```

Reply:
I'll do these cleanups in a separate PR.

Reply:
I have clippy cleanups locally, so I can make that PR if you want.